entrypoint.sh

```diff
@@ -30,161 +35,183 @@ echo -e "\tINITIAL_VERSION: ${initial_version}"
 echo -e "\tTAG_CONTEXT: ${tag_context}"
 echo -e "\tPRERELEASE_SUFFIX: ${suffix}"
 echo -e "\tVERBOSE: ${verbose}"
+echo -e "\tGITHUB_WORKSPACE: ${GITHUB_WORKSPACE}"
+
+prefix=""
+if ${with_v}
+then
+    prefix="v"
+fi
 
 current_branch=$(git rev-parse --abbrev-ref HEAD)
 
 pre_release="true"
 IFS=',' read -ra branch <<< "$release_branches"
 for b in "${branch[@]}"; do
-    echo "Is $b a match for ${current_branch}"
-    if [[ "${current_branch}" =~ $b ]]
+    echo "Is ${b} a match for ${current_branch}"
+    if [[ "${current_branch}" =~ ${b} ]]
     then
         pre_release="false"
     fi
 done
-echo "pre_release = $pre_release"
+echo "pre_release = ${pre_release}"
 
 # fetch tags
 git fetch --tags
-    
-tagFmt="^v?[0-9]+\.[0-9]+\.[0-9]+$" 
-preTagFmt="^v?[0-9]+\.[0-9]+\.[0-9]+(-$suffix\.[0-9]+)?$" 
+
+tagFmt="^${prefix}[0-9]+\.[0-9]+\.[0-9]+$"
+preTagFmt="^${prefix}[0-9]+\.[0-9]+\.[0-9]+(-${suffix}\.[0-9]+)?$"
```

Comment:
I would just like to confirm, have you tested the use where you start without prefix, then use a v prefix, and finally revert to not using a prefix?
E.g. 
v1.2.0 => 1.3.0 (with init value 1.3.0)=> v1.4.0 (with init value 1.4.0)?
I believe with this code it would do:
v1.2.0 => 1.3.0 (with init value 1.3.0)=> v1.3.0 (with init value 1.4.0) because on the third time, a latest tag with the prefix would be found (v1.2.0) and would thus increase it discarding the provided initial value.

Reply:
You are right this code won't work if you change back and forth the `WITH_V` option, it would only work if you change the option only once in the whole lifetime of a repository.
To be honest I'm not sure that changing this option several time during the lifetime of a repo is done very often and a must requirement. I think if you want to be able to do this, then you must revert the use of the `semver` tool otherwise the script will be very complicated for such edge case. The main issue with `semver` is that it accept version with and/or without `v` prefix but it always strip the `v` prefix in its output.

Reply:
Hey guys. Happy Monday! 🙋🏾‍♂️

Just wondering what the outcome of this is? Still testing the edge case or are we good with it?